### ISSUE TITLE
aria_content_in_landmark flags skip links as content outside landmark

### ISSUE THREAD

**Issue by stjordan** [IBMa/equal-access#2134]
**Describe the bug**
The `aria_content_in_landmark` rule flags skip links as content outside a landmark. Skip links are not content and adding a needless landmark worsens the screen reader experience. 

**To Reproduce**
1. Go to https://designsystem.digital.gov/ or many other websites, such as whitehouse.gov
2. Run the Accessibility Assessment from Chrome Dev Tools
3. Observe that `/document[1] /link[1]` is flagged for being outside a landmark

**Expected behavior**
`/document[1] /link[1]` is not flagged for being outside a landmark. Perhaps the heuristic is to exclude the link from this rule if it is the first link in the DOM and its target is the main landmark. 

**Desktop (please complete the following information):**
 - OS: macOS 14.6
 - Browser: chrome
 - Version: 131.0.6778.139

## Definition of Done

- [ ] Design artifacts reviewed
- [ ] Work item implemented
- [ ] Automated test case created (when applicable)
- [ ] Unit test completed
- [ ] Run Checker (when applicable)
- [ ] Manual and Screen Reader Testing (when applicable)
- [ ] Video recording of user experience walkthrough (if impact to UX)
- [ ] PR created

**Comment by tombrunet** [comment]
@stjordan This rule is not included in the WCAG rulesets - it is only in the IBM Accessibility ruleset. By convention, the skip link would be the first link in the banner, and therefore should be within a landmark. This is working as designed.

**Comment by philljenkins** [comment]
Here is an example of the skip link being inside a header:
https://react.carbondesignsystem.com/iframe.html?globals=&id=components-ui-shell-header--header-w-navigation-actions-and-side-nav&viewMode=story
